en/docs/genai/develop/components/data-loaders.md

```
@@ -0,0 +1,108 @@
+| **SharePoint Connection Configurations** | Yes | The authentication and service configuration shared by all sources. See [Connection configurations](#connection-configurations). |
+| **Data Sources** | Yes | One or more SharePoint sources to load documents from. At least one is required. See [Data sources](#data-sources). |
+| **Data Loader Name** | Yes | The variable name for the loader instance. |
+| **Result Type** | Yes | The variable type, set to `sharepoint:TextDataLoader`. |
+
+### Connection configurations
+
+The connection configuration is shared by every source.
+
+| Field | Type | Default | Description |
+|---|---|---|---|
+| **auth** | `OAuth2ClientCredentialsGrantConfig \| OAuth2RefreshTokenGrantConfig \| http:BearerTokenConfig` | — | Authentication configuration. See [Authentication](#authentication). |
```

markdownlint-cli2 (0.22.1)

[warning] Link fragments should be valid

(MD051, link-fragments)